### PR TITLE
Upmerge 2020 09 16

### PR DESCRIPTION
**Pull request by tejlmand**
sdk-nrf: https://github.com/nrfconnect/sdk-nrf/pull/2918

### CODE REVIEW

**Review by frkv — APPROVED** [review]
LGTM

File: mpsl/Kconfig [hunk 1]
```diff
@@ -10,6 +10,7 @@ config MPSL
 	select ZERO_LATENCY_IRQS
 	select IEEE802154_NRF5_EXT_IRQ_MGMT if IEEE802154_NRF5
 	select NRF_802154_MULTIPROTOCOL_SUPPORT if NRF_802154_RADIO_DRIVER
+	select MULTITHREADING_LOCK
```
Comment (rugeGerritsen) [inline mpsl/Kconfig:13]: @tejlmand , please split this out to a separate commit

Comment (tejlmand) [reply mpsl/Kconfig:13]: To my knowledge, changes required because of an upmerge should go together, otherwise bisectability will be broken.
Two commits will mean that builds are broken in between.

Comment (tejlmand) [reply mpsl/Kconfig:13]: commit message updated.

**Comment by MarekPieta** [comment]
The `BT_LL_SOFTDEVICE` Kconfig option must select `BT_CTLR_PHY_2M_SUPPORT`  and `BT_CTLR_PHY_CODED_SUPPORT`. These  are new Kconfig options related to Zephyr's Bluetooth LL configuration. Without these options the 2M PHY is not used and devices cannot switch to LLPM connection parameters (problems were replicated on nrf_desktop devices).
@tejlmand, do you want to add this as part of your commit or should I add this in separate commit on this PR?
@rugeGerritsen, please verify if all the other supported features are already added

**Comment by rugeGerritsen** [comment]
> The `BT_LL_SOFTDEVICE` Kconfig option must select `BT_CTLR_PHY_2M_SUPPORT` and `BT_CTLR_PHY_CODED_SUPPORT`. These are new Kconfig options related to Zephyr's Bluetooth LL configuration. Without these options the 2M PHY is not used and devices cannot switch to LLPM connection parameters (problems were replicated on nrf_desktop devices).
> @tejlmand, do you want to add this as part of your commit or should I add this in separate commit on this PR?
> @rugeGerritsen, please verify if all the other supported features are already added

You are right. `BT_CTLR_PHY_2M_SUPPORT` should be added under `BT_LL_SOFTDEVICE`. `BT_CTLR_PHY_CODED_SUPPORT` should be added under `SOFTDEVICE_CONTROLLER_S140`

**Comment by tejlmand** [comment]
>  @tejlmand, do you want to add this as part of your commit or should I add this in separate commit on this PR?

feel free to add separate commit, and I can squash them.
makes it easier to keep track of changes during upmerge fixes.

File: mpsl/Kconfig (hunk 1 above)
Comment (joerchan) [reply mpsl/Kconfig:13]: @tejlmand I think the update to the manifest is what counts as it will pull all the new changes in. So it can be two commits. Maybe @carlescufi can elaborate?

**Review by joerchan — APPROVED** [review]

**Review by rugeGerritsen — APPROVED** [review]

File: mpsl/Kconfig (hunk 1 above)
Comment (carlescufi) [reply mpsl/Kconfig:13]: No, it has to be a single commit if we are to keep this bisectable @rugeGerritsen and @joerchan.
